Sources/AblyChat/Rooms.swift

```diff
@@ -21,8 +21,55 @@ internal actor DefaultRooms<RoomFactory: AblyChat.RoomFactory>: Rooms {
     private let logger: InternalLogger
     private let roomFactory: RoomFactory
 
-    /// The set of rooms, keyed by room ID.
-    private var rooms: [String: RoomFactory.Room] = [:]
+    /// All the state that a `DefaultRooms` instance might hold for a given room ID.
+    private enum RoomState {
+        /// There is no room map entry (see ``RoomMapEntry`` for meaning of this term) for this room ID, but a CHA-RC1g release operation is in progress.
+        case releaseOperationInProgress(releaseTask: Task<Void, Never>)
```

Comment:
Why room can't be kept in the map until it fully released so you don't need additional enum and logic associated with it?

Reply:
The “room map” is a concept of the spec, and the spec is prescriptive about what room map manipulations should be performed, so I didn’t want to mess with that.